src/Moryx.Drivers.Mqtt/MqttTopic.cs

```diff
@@ -348,52 +409,85 @@ internal override Task OnSend(object payload)
             }
 
             topic = MqttDriver.Identifier + topic;
-            return MqttDriver.OnSend(new MqttMessageTopic(ResponseTopic, topic), msg);
+            return MqttDriver.OnSend(new MqttMessageTopic(ResponseTopic, topic, payload is IRetainAwareMessage ram ? ram.Retain ?? Retain : Retain), msg);
         }
 
-        internal override void OnReceived(string receivedTopic, byte[] messageAsBytes)
+        internal override void OnReceived(string receivedTopic, ArraySegment<byte> messageAsBytes, string? responseTopic, bool retain)
         {
-            if (MessageType != null)
+            TMessage msg;
```

Comment:
Is it intended that you have the Recieve Activity in the driver and an extra parsing here?